native/src/main/java/io/ballerina/stdlib/sql/nativeimpl/ExecuteProcessor.java

```diff
@@ -135,7 +136,11 @@ private static Object nativeExecuteExecutable(BObject client, BObject paramSQLSt
             } catch (ApplicationError e) {
                 return ErrorGenerator.getSQLApplicationError(e);
             } catch (Throwable th) {
-                return ErrorGenerator.getSQLError(th, String.format("Error while executing SQL query: %s. ", sqlQuery));
+                if (th instanceof BError) {
```

Comment:
We can catch BError is I am not mistaken.

Reply:
Removed this as no need this check when throwing an exception.